app/services/customers/eu_auto_taxes_service.rb

```diff
@@ -61,5 +65,13 @@ def applicable_tax_exceptions(country_code:)
     def eu_country_exceptions(country_code:)
       @eu_country_exceptions ||= LagoEuVat::Rate.new.country_rates(country_code:)[:exceptions]
     end
+
+    def should_apply_eu_taxes?
+      return false unless customer.organization.eu_tax_management
+
+      non_existing_eu_taxes = customer.taxes.where('code ILIKE ?', 'lago_eu%').none?
+
+      new_record || non_existing_eu_taxes || changed_attributes
```

Comment:
I'm not too sure about this condition. If the record is new but the country doesn't have eu taxes, it would return `true`. I'd be expecting that if the country has no eu_tax, it doesn't matter if it's new or attribute has changed?

Maybe this could be 2 different checks 🤔 

```ruby
def call
  return result unless should_apply_eu_taxes?
  return result if customer.taxes.where('code ILIKE ?', 'lago_eu%').none?

  #...
end

def should_apply_eu_taxes?
  return false unless customer.organization.eu_tax_management
  new_record || changed_attributes
end
```

What do you think?